src/executor/mod.rs

```diff
@@ -1,5 +1,7 @@
 //! Types which can take a boxed closure and execute it, preferably in the background.
 
+#![cfg_attr(feature="clippy", allow(boxed_local))]
```

Comment:
Seems like a false positive, might be a good idea to report it if it hasn't been already, or does `exec(self) where Self: Sized` work?